### DISCUSSION TITLE
Alter __str__ of model in custom script

### DISCUSSION THREAD

**Discussion by lanbugs** [netbox-community/netbox#7942] [Q&A]
Hi @all,

I try to build a custom script for creating VMs in one single step.

I have 3 fields at the moment:
- VM Name
- Site (Dropdown with sites)
- Prefix (depends on selected site)

I want to alter the __str__ Meta field of the ipam.prefix model that not only the prefix is shown in the dropdown, I want to print prefix + description.

```class DeployVM(Script):

    class Meta:
        name = "Deploy new VM"
        description = "Provision a new VM"

    vm_name = StringVar(description="Name of the new site")
    site = ObjectVar(description = "Site to be deployed", model=Site)
    prefix = ObjectVar(description = "Prefix", model=Prefix, query_params= {'site_id': '$site'})

    def run(self, data, commit):
        pass
```

My idea was to inherit the Prefix model and to overwrite the __str__ function in the Meta class but I struggle here:

```
class PrefixC(Prefix):
    class Meta(Prefix.Meta):
        def __str__(self):
            return "{} - {}".format(str(self.prefix), str(self.description))

class DeployVM(Script):

    class Meta:
        name = "Deploy new VM"
        description = "Provision a new VM"

    vm_name = StringVar(description="Name of the new site")
    site = ObjectVar(description = "Site to be deployed", model=Site)
    prefix = ObjectVar(description = "Prefix", model=PrefixC, query_params= {'site_id': '$site'})

    def run(self, data, commit):
        pass
```

When I did this i got an error from django:

```<class 'RuntimeError'>

Model class deploy_vm.PrefixC doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.

Python version: 3.8.10
NetBox version: 3.0.10
```

Has someone an idea how i can handle this in a smooth way?

For any help i would be appreciate,

Regards Max

**Comment by PieterL75** [comment]
Cool Idea.. I had this https://github.com/netbox-community/netbox/discussions/6586 idea posted, which kind of handles the same.
Overriding the class is maybe a more flexible way to do so.. if it is possible

**Reply by lanbugs** [reply]
Nice that I am not the only one :-)

**Comment by jeremystretch** [comment]
You should never need to override the `__str__()` method on a class. Just reference whatever particular attributes you want to display instead of relying on the object's default string representation.

**Reply by lanbugs** [reply]
Do you have an example, i don`t know what you mean.

**Reply by PieterL75** [reply]
@jeremystretch do you have an example on how to accomplish this ?
Goal is to reference the Prefix, but have the dropdown display more information than the default __str__ returns

**Reply by candlerb** [reply]
Duplicate of: #6018, #6586 ~~and with useful info [here](https://groups.google.com/g/netbox-discuss/c/5F_DAmkjIbc/m/_LzzA21zBAAJ)~~.

~~In short: if what you want to display is available as an attribute on the model, then you can set `display_field='attrname'` in the ObjectVar.~~ *(EDIT: that option was removed in Netbox 3.0)*  As far as I can tell you are SOL, unless you are prepared to pre-generate the complete list as a ChoiceVar.
